File: cpp/leetcode-indexOfFirstSubstring.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.length() > haystack.length()) {
            return -1;
        }

        for (int i = 0; i < haystack.length(); i ++) {
            for (int k = 0; k < needle.length(); k++) {
                if (k + i >= haystack.length() || haystack[i + k] != needle[k]) break;
                if (k == needle.length() - 1) return i;
            }
        }
        return -1;
    }
};

File: rs/leetcode-indexOfFirstSubstring.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        if needle.len() > haystack.len() {return -1;}

        for i in 0..haystack.len() {
            for k in 0..needle.len() {
                if i + k >= haystack.len() || haystack.as_bytes()[i + k] != needle.as_bytes()[k] {break;}
                if k == needle.len() - 1 {return i as i32;}
            }
        }
        -1
    }
}